OADP-5634: Add new structure for 3scale user story

apurvabhide17 opened the pull request:
Version(s):
OCP 4.14 → OCP 4.19

Issue:
[OADP-5634](https://issues.redhat.com/browse/OADP-5634)

Link to docs preview:

- [Backing up and restoring 3scale API Management by using OADP
](https://89068--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-api-management-by-using-oadp)
- [Backing up 3scale API Management by using OADP](https://89068--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-api-management-by-using-oadp)
- [Restoring 3scale API Management by using OADP](https://89068--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-api-management-by-using-oadp)

QE review:
- [x] QE has approved this change.

Additional information - 
This PR focuses on the following points to improve the existing [Backing up and restoring 3scale by using OADP](https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.html). user story from the related PR - [PR#85994](https://github.com/openshift/openshift-docs/pull/85994) - 

- Updating description/command for oc get deployment -n threescale and oc edit deployment system-mysql -n threescale.

Structural changes to split Backing up and restoring 3scale by using OADP:

- Backing up 3scale
-- Creating the Data Protection Application
-- Backing up the 3scale Operator
-- Backing up the mysql database
-- Backing up the backend Redis database

- Restoring 3scale
-- Restoring the secrets and APIManager
-- Restoring the mysql database
-- Restoring the back-end Redis database
-- Scaling up the 3scale operator and deployment

apurvabhide17 committed:
OADP-5634: Add new structure for 3scale user story

QE review

Peer review changes

Changes related to 3scale usage

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,24 @@
+.Additional resources
+* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-creating-backup-cr.adoc#oadp-creating-backup-cr-doc[Creating a Backup CR]
+
+include::modules/backing-up-the-mysql-database.adoc[leveloffset=+1]
```
```suggestion
include::modules/backing-up-the-mysql-database.adoc[leveloffset=+2]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,24 @@
+
+include::modules/backing-up-the-mysql-database.adoc[leveloffset=+1]
+
+include::modules/backing-up-the-backend-redis-database.adoc[leveloffset=+1]
```
```suggestion
include::modules/backing-up-the-backend-redis-database.adoc[leveloffset=+2]
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,24 @@
+[role="_additional-resources"]
+.Additional resources
+* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-creating-backup-cr.adoc#oadp-creating-backup-cr-doc[Creating a Backup CR]
+
```
```suggestion
Add a *Back up use cases* section 
```

apurvabhide17 replied:
@anarnold97, could you please help me by sharing a few examples for this?

anarnold97 replied:
@stillalearner - please can i ask if you have any backup test cases you could share with Apurva?
Thanks

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,24 @@
+
+toc::[]
+
+You can back up 3scale by backing up 3scale Operator, and databases such as MySQL and Redis.
```
```suggestion
You can back up 3scale by backing up 3scale Operator. You can also backup and databases such as MySQL and Redis.
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -15,29 +19,9 @@ In this example, the non-service affecting approach is used to back up and resto
-* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc#oadp-installing-dpa_installing-oadp-aws[Installing the Data Protection Application]
-
-include::modules/backing-up-the-3scale-operator.adoc[leveloffset=+1]
+You can xref:../../../backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc#backing-up-3scale-by-using-oadp[back up 3scale] and xref:../../../backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-by-using-oadp.adoc#restoring-3scale-by-using-oadp[restore 3scale] by using OADP.
```
this would be better split into two lines, possibly bullet-pointed

![image](https://github.com/user-attachments/assets/ecedc0b3-8a88-4abe-8e08-3cb4bb5a66a3)

think of user navigating this document in the context of this sentence

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,17 @@
+
+toc::[]
+
+You can restore the 3scale by restoring the backed up 3scale Operator resources, and databases such as MySQL and Redis. Once the data has been restored, you can scale up the 3scale Operator and deployment. 
```
```suggestion
You can restore the 3scale by restoring the backed up 3scale Operator resources. 

You can also restore databases such as MySQL and Redis. 

Once the data has been restored, you can scale up the 3scale Operator and deployment. 
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,17 @@
+include::_attributes/common-attributes.adoc[]
+:context: restoring-3scale-by-using-oadp
+
+toc::[]
```
i wonder whether you should add an introduction, then putting these use cases underneath that introduction. Even if it is just a short sentence, followed by a list of the restore use cases

dfennessy commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -1,9 +1,13 @@
+
 toc::[]
 
 With Red Hat 3scale API Management (APIM), you can manage your APIs for internal or external users. Share, secure, distribute, control, and monetize your APIs on an infrastructure platform built with performance, customer control, and future growth in mind.
```
As far as I am aware, there isn't an acronym like APIM used for Red Hat 3scale API Management. You can remove it. After the first usage of Red Hat 3scale API Management, you can use 3scale API Management thereafter.

For this, we can think in terms of our user story structured like the following:

As a [user role], I want to [perform an action], so that [benefit or goal].

For OADP, we might write the user story something like: 

As an OpenShift administrator, I want to use OpenShift API for Data Protection (OADP) to back up and restore applications, cluster resources, persistent volumes, and internal images, so that I can ensure comprehensive disaster recovery and protect both containerized applications and virtual machines (VMs) on OpenShift.

Based on the user story, we might have an intro similar to this:
Back up and restore for 3scale API Management is critical for ensuring data protection and disaster recovery. With OpenShift API for Data Protection (OADP), you can safeguard 3scale deployments by backing up application resources, persistent volumes, and configurations. You can configure OADP to create backups of 3scale components and restore them when needed, ensuring minimal downtime and data integrity.

My knowledge of OADP is minimal, so this is just an example that hopefully helps.

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -1,13 +1,22 @@
+
+With {oadp-first}, you can safeguard 3scale deployments by backing up application resources, persistent volumes, and configurations. You can configure OADP to perform the following operations with 3scale: 
+
+* create backup of 3scale components by following the steps from xref:../../../backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc#backing-up-3scale-by-using-oadp[Backing up 3scale].
```
```suggestion
* Create backup of 3scale components by following the steps from xref:../../../backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc#backing-up-3scale-by-using-oadp[Backing up 3scale].
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -1,13 +1,22 @@
+With {oadp-first}, you can safeguard 3scale deployments by backing up application resources, persistent volumes, and configurations. You can configure OADP to perform the following operations with 3scale: 
+
+* create backup of 3scale components by following the steps from xref:../../../backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc#backing-up-3scale-by-using-oadp[Backing up 3scale].
+* restore the components to scale up the 3scale Operator and deployment by following the steps from xref:../../../backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-by-using-oadp.adoc#restoring-3scale-by-using-oadp[Restoring 3scale].
```
```suggestion
* Restore the components to scale up the 3scale Operator and deployment by following the steps from xref:../../../backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-by-using-oadp.adoc#restoring-3scale-by-using-oadp[Restoring 3scale].
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -1,13 +1,22 @@
-With Red Hat 3scale API Management (APIM), you can manage your APIs for internal or external users. Share, secure, distribute, control, and monetize your APIs on an infrastructure platform built with performance, customer control, and future growth in mind.
-You can deploy 3scale components on-premise, in the cloud, as a managed service, or in any combination based on your requirement.
+With Red Hat 3scale API Management, you can manage your APIs for internal or external users. 
+You can deploy 3scale API Management components on-premise, in the cloud, as a managed service, or in any combination based on your requirement.
```
```suggestion
You can deploy 3scale API Management components on-premise, in the cloud, as a managed service, or in any combination based on your requirements.
```

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,25 @@
+
+toc::[]
+
+You can back up 3scale by backing up 3scale Operator. You can also backup and databases such as MySQL and Redis.
```
```suggestion
You can back up 3scale by backing up 3scale Operator. You can also back up databases such as MySQL and Redis.
```
Verb is back up

apurvabhide17 replied:
my bad, while implementing you suggestion there was a typo when I added it as a separate sentence. It is now fixed.

anarnold97 commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -40,7 +40,7 @@ spec:
 ----
 
-. Attach the PVC to the system database pod by editing the system database deployment to use the `mysql` dump:
+. Attach the PVC to the system database pod by editing the `system-mysql` deployment to use the `mysql` dump:
```
The formatting in this file is not correct:

![image](https://github.com/user-attachments/assets/83e781f7-5690-4219-b86a-fa1ea499be39)

I think it should be MySQL not mysql

![image](https://github.com/user-attachments/assets/6d2cead0-393f-4972-bc3b-dbdfd501a13d)

apurvabhide17 replied:
I've replaced `mysql` with MySQL in all the modules.

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,25 @@
+.Additional resources
+* xref:../../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc#oadp-installing-dpa_installing-oadp-aws[Installing the Data Protection Application]
+
+include::modules/backing-up-the-3scale-operator.adoc[leveloffset=+1]
```
In this module why is a **Use case** listed as **Next steps**

Do you think it looks a bit weird?

apurvabhide17 replied:
As per the user story template that we had created for OADP, I've added the module under next steps to explain the sequence to the users. 
Let me know if you've any other suggestion for the "Next steps" section, or if it is not required.

anarnold97 replied:
Please remove *Next steps*. 
You are talking about the next use cases and not logical next steps.

apurvabhide17 replied:
done

anarnold97 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,18 @@
+
+include::modules/restoring-the-secrets-and-apimanager.adoc[leveloffset=+1]
+
+include::modules/restoring-the-mysql-database.adoc[leveloffset=+1]
```
![image](https://github.com/user-attachments/assets/0324e494-cc48-4464-a7e0-fd4821cf8d95)
 Same here

anarnold97 replied:
Is the next step after restoring the MySQL DB,  backing up Redis DB? or are they independent use case elements?

apurvabhide17 replied:
After [restoring mysql](https://89068--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-3scale/restoring-3scale-by-using-oadp), the next step is restoring the backend-redis database followed by scaling 3scale operator.
For [backing up 3scale](https://89068--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp), the sequence is - backing up 3scale operator followed by backing up mysql database and then redis database.

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,25 @@
+You can back up 3scale by backing up 3scale Operator. You can also backup and databases such as MySQL and Redis.
+
+
+include::modules/creating-the-data-protection-application.adoc[leveloffset=+1]
```
Yes, it was suggested in the OCP review for using xref as adding it in the module isn't supported, and the same additional resources related information was added in the assembly. 
Let me know if you have any other suggestion for this approach.

anarnold97 replied:
I would remove - `For more information ... " . Without an xref, it just looks weird like it should be a link but is mis-formatted
WDYT

If it is so important, add an additional resources section at the bottom of the document

apurvabhide17 replied:
I don't mind removing that line and having the existing xref added to additional resources section in the assembly.

shdeshpa07 commented on modules/restoring-the-backend-redis-database.adoc:
```diff
@@ -10,7 +10,7 @@ You can restore the back-end Redis database by deleting the deployment and speci
 .Prerequisites
```
```suggestion
You can restore the back-end Redis database by deleting the deployment and specifying the resources you do not want to restore.
```

apurvabhide17 replied:
There is a step added in the procedure explaining how to delete it, and is not a prerequisite.

shdeshpa07 commented on modules/restoring-the-backend-redis-database.adoc:
```diff
@@ -10,7 +10,7 @@ You can restore the back-end Redis database by deleting the deployment and speci
-* You restored the `mysql` database.
+* You restored the MySQL database.
 
 .Procedure
```
Call out the `backupName: redis-backup` at step 2 of the procedure. User needs to know that they are restoring the redis backup.

shdeshpa07 commented on modules/scaling-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -22,11 +22,11 @@ There might be some `system-mysql` or `backend-redis` pods running detached from
+$ oc get pods -n threescale
 ----
 
 . Scale up the deployments by executing the following script:
```
```suggestion
. Scale up the deployments by running the following script:
```

shdeshpa07 replied:
Avoid the word `execute`.

shdeshpa07 replied:
In the last step, step 5, rewrite the sentence:

. Use the URL from this output to log in to the 3scale Operator as an administrator. You can verify that the data, when you took the backup, is available.

shdeshpa07 reviewed (commented):
Left a few comments. I have a few suggestions for consistency with other OADP doc sets. Please let me know if you have questions. Thanks!

apurvabhide17 commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -3,13 +3,13 @@
 * The `Pod`, `ReplicationController`, and `Deployment` objects.
 * The additional persistent volumes (PVs) and associated persistent volume claims (PVCs). 
-* The `mysql` dump, which the `example-claim` PVC contains.
+* The MySQL dump, which the `example-claim` PVC contains.
```
As per IBM SSG - Use "which", preceded by a comma, to introduce a nonrestrictive clause. Use "that", without a comma, to introduce a
restrictive clause.

apurvabhide17 commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -11,7 +11,7 @@ You can back up the Redis database by adding the required annotations and by lis
+* You backed up the MySQL database.
 * The Redis queues have been drained before performing the backup.
 
 
```
I'll check it again with Sachin but the previous command opens the backend-redis deployment, where this annotation must be added.

apurvabhide17 replied:
No need to add the resources because these steps will only edit the existing resources to add annotation.

apurvabhide17 commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -11,7 +11,7 @@ You can back up the Redis database by adding the required annotations and by lis
 .Prerequisites
 
```
This is not a prerequisite.

apurvabhide17 commented on modules/backing-up-the-3scale-operator.adoc:
```diff
@@ -5,7 +5,7 @@
 = Backing up the 3scale Operator
 
-You can back up the Operator resources, and Secret and APIManager custom resources (CR). For more information, see "Creating a Backup CR".
+You can back up the Operator resources, and Secret and APIManager custom resources (CR).
```
I've added it for Secret but not for APIManager. In 3scale docs, they are not using back ticks for APIManager - https://docs.redhat.com/en/documentation/red_hat_3scale_api_management/2.15/html/installing_red_hat_3scale_api_management/index, following the same style.

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -17,27 +26,5 @@ Additionally, ensure that you are restoring 3scale on the same cluster where it
-include::modules/restoring-the-mysql-database.adoc[leveloffset=+1]
-
-include::modules/restoring-the-backend-redis-database.adoc[leveloffset=+1]
+* You installed and configured Red Hat 3scale. For more information, see link:{Link3scaleInstalling3scale}#install-threescale-on-openshift-guide[Installing 3scale API Management on OpenShift] and link:{Link3scaleInstalling3scale}[Red Hat 3scale API Management].
```
Added it to the backup and restore assembly instead of adding it here.

anarnold97 commented:
LGTM ... no need to send it for another review, once QE are happy kick it over to OCP review. 
 
Taxonomy looks good, easier to navigate

I think this works well

![image](https://github.com/user-attachments/assets/16da0cbf-0201-42ed-b60d-bd41c4efeb77)

Eases navigation... should we do more things like this?

apurvabhide17 commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -91,7 +91,7 @@ Warning: apps.openshift.io/v1 deployment is deprecated in v4.14+, unavailable in
 ----
 
-. Create the following YAML file to restore the `mysql` database:
+. Create the following YAML file to restore the MySQL database:
```
Added about it to both backup and restore file.

apurvabhide17 commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -11,7 +11,7 @@ You can restore the Secrets and APIManager by using the following procedure.
+* You backed up MySQL and Redis databases.
 * You are restoring the database on the same cluster, where it was backed up. 
 +
 If it is on a different cluster, install and configure {oadp-short} with `nodeAgent` enabled on the destination cluster as it was on the source cluster.
```
Added about it to both backup and restore file.

apurvabhide17 commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -11,7 +11,7 @@ You can restore the Secrets and APIManager by using the following procedure.
 * You backed up the 3scale Operator.
-* You backed up `mysql` and Redis databases.
+* You backed up MySQL and Redis databases.
 * You are restoring the database on the same cluster, where it was backed up. 
```
Added about it to both backup and restore file.

stillalearner commented on modules/scaling-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -21,15 +21,28 @@ There might be some `system-mysql` or `backend-redis` pods running detached from
+. Scale up the deployments by running the following script:
 +
 [source,terminal]
 ----
```
where are the contents of this script?

apurvabhide17 replied:
@stillalearner, could you share the content of this script? 
Or should it be same like "scaledowndeployment.sh"?

apurvabhide17 replied:
done, I've added it.

stillalearner commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -69,23 +67,40 @@ spec:
 +
 [source,terminal]
 ----
 $ oc get backups.velero.io
```
oc get backups.velero.io redis-backup

stillalearner commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -69,23 +67,40 @@ spec:
-* Restore the Secrets and APIManager CRs.
\ No newline at end of file
++
+.Example output:
```
and this output needs to be changed according to command.

stillalearner commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -6,15 +6,15 @@
 [id="restoring-the-secrets-and-apimanager_{context}"]
 = Restoring the secrets and APIManager
```
Restoring the 3scale operator resources, secrets and APIManager

apurvabhide17 commented:
/test validate-asciidoc
/test validate-portal

stillalearner commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -69,23 +67,36 @@ spec:
++
+.Example output:
++
+[source,terminal]
```
please paste the output after running -o yaml
like this
----

status:
  completionTimestamp: "2025-04-17T13:25:19Z"
  errors: 1
  expiration: "2025-05-17T13:25:16Z"
  formatVersion: 1.1.0
  hookStatus: {}
  phase: Completed
  progress: {}
  startTimestamp: "2025-04-17T13:25:16Z"
  version: 1

apurvabhide17 replied:
done, I've updated the output

stillalearner commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -111,19 +111,27 @@ spec:
+* Verify that the MySQL backup is completed by running the following command:
 +
 [source,terminal]
 ----
```
same here -o yaml

and same output

apurvabhide17 replied:
done, I've updated the output

apurvabhide17 commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -69,23 +67,36 @@ spec:
 [source,terminal]
 ----
-$ oc get backups.velero.io
+$ oc get backups.velero.io redis-backup
```
done

stillalearner reviewed (approved):
lgtm

apurvabhide17 commented:
/label peer-review-needed

ShaunaDiaz commented:
@apurvabhide17 This PR cannot merge into `main` unless it also goes to `enterprise-4.19`.  If this content _cannot_ also go in 4.19, you can make a new one from the 4.18 branch and we can cherry-pick back. Please confirm either way.

apurvabhide17 commented:
> @apurvabhide17 This PR cannot merge into `main` unless it also goes to `enterprise-4.19`. If this content _cannot_ also go in 4.19, you can make a new one from the 4.18 branch and we can cherry-pick back. Please confirm either way.

Hi @ShaunaDiaz, I've updated the version after confirming it with QE.

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -3,18 +3,18 @@
+= Restoring the 3scale Operator, secrets, and APIManager
 
-You can restore the Secrets and APIManager by using the following procedure.
+You can restore the 3scale Operator resources, and `Secret` and APIManager custom resources (CR) by using the following procedure.
```
```suggestion
You can restore the Red{nbsp}Hat build of 3scale Operator and `Secret` and APIManager custom resources (CRs) by using the following procedure.
```
why?

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -3,18 +3,18 @@
 
 .Prerequisites
 
 * You backed up the 3scale Operator.
```
```suggestion
* You backed up the Red{nbsp}Hat build of 3scale Operator.
```
I would fix this naming throughout.

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -3,18 +3,18 @@
 
 * You backed up the 3scale Operator.
-* You backed up `mysql` and Redis databases.
+* You backed up the MySQL and Redis databases.
```
```suggestion
* You backed up the MySQL and Redis databases.
```
Again, are these required databases, supported databases, just examples for any customer databases?

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -3,18 +3,18 @@
 * You are restoring the database on the same cluster, where it was backed up. 
 +
-If it is on a different cluster, install and configure {oadp-short} with `nodeAgent` enabled on the destination cluster as it was on the source cluster.
+If you are restoring on a different cluster, install and configure {oadp-short} with `nodeAgent` enabled on the destination cluster. Ensure that the {oadp-short} configuration is same as it was on the source cluster.
```
```suggestion
If you are restoring the Operator to a different cluster that you backed up from, install and configure {oadp-short} with `nodeAgent` enabled on the destination cluster. Ensure that the {oadp-short} configuration is same as it was on the source cluster.
```

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -57,15 +57,23 @@ spec:
   - backuprepositories.velero.io
   itemOperationTimeout: 4h0m0s
 ----
+<1> Restoring the 3scale Operator's backup
```
? this clearly needs some explanatory text

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -95,8 +103,15 @@ EOF
+----
 
-. Create a YAML file with the following configuration to restore the Secrets:
+. Create a YAML file with the following configuration to restore the `Secret`:
```
```suggestion
. Create a `restore-secret` YAML file with the following configuration to restore the `Secret` CR:
```

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -121,13 +136,21 @@ spec:
   - backuprepositories.velero.io
   itemOperationTimeout: 4h0m0s
 ----
+<1> Restoring the `Secret` backup.
```
explanatory text, e.g., "Use the name you specified in the `metadata.name` field of the backup YAML", etc.

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -121,13 +136,21 @@ spec:
+<1> Restoring the `Secret` backup.
 
-. Restore the Secrets by running the following command:
+. Restore the `Secret` by running the following command:
```
```suggestion
. Restore the `Secret` CR by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -153,22 +176,33 @@ spec:
 ----
-<1> The resources that you do not want to restore.
+<1> Restoring the APIManager backup.
+<2> The resources that you do not want to restore.
```
```suggestion
<2> List any resources that you do not want to restore.
```

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -153,22 +176,33 @@ spec:
   itemOperationTimeout: 4h0m0s
 ----
-<1> The resources that you do not want to restore.
+<1> Restoring the APIManager backup.
```
explanatory text needed

ShaunaDiaz commented on modules/restoring-the-3scale-operator-secrets-and-apimanager.adoc:
```diff
@@ -153,22 +176,33 @@ spec:
+<1> Restoring the APIManager backup.
+<2> The resources that you do not want to restore.
 
 . Restore the APIManager by running the following command:
```
```suggestion
. Restore the `APIManager` CR by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-backend-redis-database.adoc:
```diff
@@ -55,13 +55,21 @@ spec:
   itemOperationTimeout: 1h0m0s
   restorePVs: true
 ----
+<1> Restoring the Redis backup.
```
explanatory text needed

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -3,13 +3,13 @@
 
 [id="restoring-the-mysql-database_{context}"]
-= Restoring the mysql database
+= Restoring the MySQL database
```
```suggestion
= Restoring a MySQL database
```

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -3,13 +3,13 @@
+= Restoring the MySQL database
 
-Restoring the `mysql` database re-creates the following resources:
+Restoring the MySQL database re-creates the following resources:
```
```suggestion
Restoring a MySQL database re-creates the following resources:
```

ShaunaDiaz commented on modules/scaling-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -9,6 +9,8 @@ You can scale up the 3scale Operator and any deployment that was manually scaled
 .Prerequisites
 
+* You restored the 3scale Operator resources, `Secret` and APIManager custom resources (CR).
+* You restored the MySQL and back-end Redis database.
```
```suggestion
* You restored the MySQL and back-end Redis databases.
```

ShaunaDiaz reviewed (commented):
I think you need to determine the product or component naming that should be used. A lot of places need solid explanatory text that is missing; this PR feels unfinished and not truly ready for peer review. Please make changes and put back in peer review prior to submitting for merge review.

apurvabhide17 commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -11,7 +11,7 @@ You can back up the Redis database by adding the required annotations and by lis
 
 * You backed up the 3scale Operator.
-* You backed up the mysql database.
+* You backed up the MySQL database.
```
yes

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -1,43 +1,33 @@
 
-With Red Hat 3scale API Management (APIM), you can manage your APIs for internal or external users. Share, secure, distribute, control, and monetize your APIs on an infrastructure platform built with performance, customer control, and future growth in mind.
-You can deploy 3scale components on-premise, in the cloud, as a managed service, or in any combination based on your requirement.
+With Red Hat 3scale API Management, you can manage your APIs for internal or external users. 
```
that's how 3scale refers in their documentation, and APIManager is a CR

apurvabhide17 commented:
> I think you need to determine the product or component naming that should be used. A lot of places need solid explanatory text that is missing; this PR feels unfinished and not truly ready for peer review. Please make changes and put back in peer review prior to submitting for merge review.

Hi @ShaunaDiaz, thanks for reviewing my PR, I've implemented the suggestions apart from naming guidelines for 3scale. 
I had discussion with the 3scale writer and I've updated it as per the 3scale team's guidelines w.r.t to this [epic](https://issues.redhat.com/browse/THREESCALE-10508).  This includes -

- 3scale API Management instead of 3scale in the title
- Red Hat 3scale API Management in the abstract
- Once the above usage is completed, further refer to it as only 3scale for the ease of reading
- Replace Operator by operator
- There were no guidelines to use Red Hat build of 3scale
- It is APIManager and not API manager.
- It is API Management and not API management

If all looks good, could I go ahead with the merge review? If there are any other improvements, I can create another PR for it. Thanks!

apurvabhide17 commented:
/label merge-review-needed

apurvabhide17 commented:
I've incorporated all the basic suggestions related to links, callouts, prerequisites, and other suggestions from the peer-review. Apart from that, I've also followed the guidelines of using 3scale in the docs as per the [official RH product names](https://docs.google.com/spreadsheets/d/1DLS_lS3VKidgZIvcLmLp9BoiqptkvqHWfe1D5FD2kfk/edit?gid=1259317633#gid=1259317633) sheet and by discussing with the 3scale writer.

Could you please merge this PR? If there are any other suggestions, I can create another PR for it.
Thanks!

stevsmit reviewed (commented):
This looks really good overall, and nothing I've commented should hold up a merge. I've left a few comments for you to consider in a future PR. One thing I've noticed is that, in our docs, we sometimes either capitalize 3scale Operator and in your PR you do not. Maybe the official name isn't capitalized, or maybe it is. Generally for OCP we require Operator to be capitalized, but there needs to be some consistency between what's in the docs now (whether that requires changed it to a capital O or lowercase o). 

Overall, good work!

stevsmit commented:
/cherry-pick enterprise-4.19

stevsmit commented:
/cherry-pick enterprise-4.17

stevsmit commented:
/cherry-pick enterprise-4.16

stevsmit commented:
/cherry-pick enterprise-4.14

apurvabhide17 commented on _topic_maps/_topic_map.yml:
```diff
@@ -3658,8 +3658,12 @@ Topics:
     Topics:
-    - Name: Backing up and restoring 3scale by using OADP
-      File: backing-up-and-restoring-3scale-by-using-oadp
+    - Name: Backing up and restoring 3scale API Management by using OADP
```
It helps to find the sub-assemblies in the right side ToC with the existing OCP structure

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-3scale-api-management-by-using-oadp.adoc:
```diff
@@ -0,0 +1,31 @@
+
+toc::[]
+
+You can back up Red{nbsp}Hat 3scale API Management components by backing up the 3scale operator, and databases such as MySQL and Redis.
```
Since this user story focuses on 3scale, their team suggested to use "operator" instead of "Operator" along with other guidelines for using the official product name for 3scale

apurvabhide17 commented:
> This looks really good overall, and nothing I've commented should hold up a merge. I've left a few comments for you to consider in a future PR. One thing I've noticed is that, in our docs, we sometimes either capitalize 3scale Operator and in your PR you do not. Maybe the official name isn't capitalized, or maybe it is. Generally for OCP we require Operator to be capitalized, but there needs to be some consistency between what's in the docs now (whether that requires changed it to a capital O or lowercase o).
> 
> Overall, good work!

Thanks @stevsmit for reviewing and merging this PR. 
Since this user story focuses on 3scale, their team suggested to use "operator" instead of "Operator" along with other guidelines for using the official product name for 3scale
